src/core/Akka.Streams/KillSwitch.cs

```diff
@@ -179,25 +176,21 @@ public Logic(Task terminationSignal, UniqueBidiKillSwitchStage<TIn1, TOut1, TIn2
 
             #endregion
 
-            public static UniqueBidiKillSwitchStage<TIn1, TOut1, TIn2, TOut2> Instance { get; } =
-                new UniqueBidiKillSwitchStage<TIn1, TOut1, TIn2, TOut2>();
+            public static UniqueBidiKillSwitchStage<TIn, TOut> Instance { get; } = new UniqueBidiKillSwitchStage<TIn, TOut>();
 
-            private UniqueBidiKillSwitchStage()
-            {
-                Shape = new BidiShape<TIn1, TOut1, TIn2, TOut2>(In1, Out1, In2, Out2);
-            }
+            private UniqueBidiKillSwitchStage() => Shape = new BidiShape<TIn, TIn, TOut, TOut>(In1, Out1, In2, Out2);
 
             protected override Attributes InitialAttributes { get; } = Attributes.CreateName("breaker");
 
-            private Inlet<TIn1> In1 { get; } = new Inlet<TIn1>("KillSwitchBidi.in1");
+            private Inlet<TIn> In1 { get; } = new Inlet<TIn>("KillSwitchBidi.in1");
 
-            private Outlet<TOut1> Out1 { get; } = new Outlet<TOut1>("KillSwitchBidi.out1");
+            private Outlet<TIn> Out1 { get; } = new Outlet<TIn>("KillSwitchBidi.out1");
 
-            private Inlet<TIn2> In2 { get; } = new Inlet<TIn2>("KillSwitchBidi.in2");
+            private Inlet<TOut> In2 { get; } = new Inlet<TOut>("KillSwitchBidi.in2");
 
-            private Outlet<TOut2> Out2 { get; } = new Outlet<TOut2>("KillSwitchBidi.out2");
+            private Outlet<TOut> Out2 { get; } = new Outlet<TOut>("KillSwitchBidi.out2");
 
-            public override BidiShape<TIn1, TOut1, TIn2, TOut2> Shape { get; }
+            public override BidiShape<TIn, TIn, TOut, TOut> Shape { get; }
```

Comment:
Are you sure, this is right? It looks like less generic version.

Reply:
Yup, the previous more generic version was actually a „bug“, only prevented from happening due to the fact that the public api already restricts it to two different types. 
I got a compilation error at the following line 
„Push(killSwitch.Out2, Grab(killSwitch.In2)),“
which makes it obvious that both, the inlets and outlets, have to be of the same type.

Reply:
The type names are a bit confusing, T1 instead of TIn and T2 instead of TOut would make more  sense, so I’ll update them when if fix the other issue with the performance test project.